### PR TITLE
[15.0][MIG] stock_analytic: migration

### PR DESCRIPTION
**Pull request by Mantux11**
Migration to `15.0`. `product` file was deleted, because in original odoo 14 code `_anglo_saxon_sale_move_lines` function was deleted. I don't know why this was left in `14.0` migration, but I say that, if no one had problems, I don't see any issues that is associated with this, so I assume that this is not necessary to leave a file that don't do anything. Improved tests.

### CODE REVIEW

**Comment by rousseldenis** [comment]
@Mantux11 Can you review this in order to integrate it here too ?

https://github.com/OCA/account-analytic/pull/423

Thanks

**Review by AaronHForgeFlow — COMMENTED** [review]
Functional LG! Pending to attend comments

**Comment by Mantux11** [comment]
@rousseldenis I'm integrated. Migration script's I guess isn't needed because you do that in `14.0` version, so I don't add that. If it's needed, say, I would insert here also.

**Comment by rousseldenis** [comment]
> @rousseldenis I'm integrated. Migration script's I guess isn't needed because you do that in `14.0` version, so I don't add that. If it's needed, say, I would insert here also.

Oops.

The good way is to cherry-pick it in order to keep authorship. Can you do it ?

And indeed, in your migration commit you can remove migration scripts.

**Comment by Mantux11** [comment]
@rousseldenis I'll wait for your PR merge, because it would be easier for me to cherry pick.

**Comment by Olageibol** [comment]
In order to maintain the data security, I believe that the analytic_account_id field in stock.move.line views needs the analytic group added to it.
https://github.com/vialaurea/OCA-account-analytic/blob/15.0-mig-stock_analytic/stock_analytic/views/stock_move_line.xml

**Comment by AaronHForgeFlow** [comment]
can you include this? https://github.com/OCA/account-analytic/pull/423

**Comment by Mantux11** [comment]
@AaronHForgeFlow yes, I would do that ASAP.

**Comment by jlzhou** [comment]
@Mantux11 thanks for your effort!

Could you please explain why remove _prepare_procurement_values override from stock.move model?

`    def _prepare_procurement_values(self):
        """
        Allows to transmit analytic account from moves to new
        moves through procurement.
        """
        res = super()._prepare_procurement_values()
        if self.analytic_account_id:
            res.update(
                {
                    "analytic_account_id": self.analytic_account_id.id,
                }
            )
        return res`

**Comment by Mantux11** [comment]
@AaronHForgeFlow I added all comitt's from 14. @jlzhou Returned this function,

**Review by AaronHForgeFlow — APPROVED** [review]
LG!

**Comment by Mantux11** [comment]
@AaronHForgeFlow @jlzhou Could you Merge this PR? It's a valid migration, and it hasn't been reviewed for more than a month.

**Comment by AaronHForgeFlow** [comment]
@Mantux11  This needs an additional approval for the merge. Perhaps you can ping the people that have commented here but did not do a formal review.

**Comment by rafaelbn** [comment]
I don't use this module. Ping my in 2 week if it not reviewed by other contributors. Thanks

File: stock_analytic/models/stock.py
Comment (Jerther) [inline stock_analytic/models/stock.py:42]: This method is not covered by unit tests

**Review by Jerther — CHANGES_REQUESTED** [review]
Missing coverage for one method. Otherwise, we've been using a previous version of this PR, identical but without this method, and we've been doing so for 2 or 3 months now and it's working fine. :)

**Review by florenciafrigieri2 — COMMENTED** [review]
Hi! I was trying to do a functional review in runboat and even though the analytic modules were installed, I was not able to allow analytic funcionalities even in the accounting app. Can you help me with that? Thanks in advance!

File: stock_analytic/models/stock.py
Comment (Mantux11) [reply stock_analytic/models/stock.py:42]: @Jerther Improved tests.

**Comment by Mantux11** [comment]
@florenciafrigieri2 It's been a long time ago, when I tested this module, so personally, I forgot. If I will find, how and where, I will update this comment.

**Review by Jerther — APPROVED** [review]
100% code coverage, and working well. It's good to go!

**Review by nicomacr — APPROVED** [review]
LGTM technical review

**Comment by Mantux11** [comment]
@rafaelbn passed 2 weeks and PR has been approved. Could you review it and merge?

**Review by rousseldenis — APPROVED** [review]
Code review

@OCA/accounting-maintainers

**Pull request merged**
